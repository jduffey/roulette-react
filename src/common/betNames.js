export const BET_NAMES = {
    STRAIGHT_UP_0: "StraightUp_0",
    STRAIGHT_UP_00: "StraightUp_00",
    STRAIGHT_UP_1: "StraightUp_1",
    STRAIGHT_UP_2: "StraightUp_2",
    STRAIGHT_UP_3: "StraightUp_3",
    STRAIGHT_UP_4: "StraightUp_4",
    STRAIGHT_UP_5: "StraightUp_5",
    STRAIGHT_UP_6: "StraightUp_6",
    STRAIGHT_UP_7: "StraightUp_7",
    STRAIGHT_UP_8: "StraightUp_8",
    STRAIGHT_UP_9: "StraightUp_9",
    STRAIGHT_UP_10: "StraightUp_10",
    STRAIGHT_UP_11: "StraightUp_11",
    STRAIGHT_UP_12: "StraightUp_12",
    STRAIGHT_UP_13: "StraightUp_13",
    STRAIGHT_UP_14: "StraightUp_14",
    STRAIGHT_UP_15: "StraightUp_15",
    STRAIGHT_UP_16: "StraightUp_16",
    STRAIGHT_UP_17: "StraightUp_17",
    STRAIGHT_UP_18: "StraightUp_18",
    STRAIGHT_UP_19: "StraightUp_19",
    STRAIGHT_UP_20: "StraightUp_20",
    STRAIGHT_UP_21: "StraightUp_21",
    STRAIGHT_UP_22: "StraightUp_22",
    STRAIGHT_UP_23: "StraightUp_23",
    STRAIGHT_UP_24: "StraightUp_24",
    STRAIGHT_UP_25: "StraightUp_25",
    STRAIGHT_UP_26: "StraightUp_26",
    STRAIGHT_UP_27: "StraightUp_27",
    STRAIGHT_UP_28: "StraightUp_28",
    STRAIGHT_UP_29: "StraightUp_29",
    STRAIGHT_UP_30: "StraightUp_30",
    STRAIGHT_UP_31: "StraightUp_31",
    STRAIGHT_UP_32: "StraightUp_32",
    STRAIGHT_UP_33: "StraightUp_33",
    STRAIGHT_UP_34: "StraightUp_34",
    STRAIGHT_UP_35: "StraightUp_35",
    STRAIGHT_UP_36: "StraightUp_36",
    FIRST_DOZEN: "1st 12",
    SECOND_DOZEN: "2nd 12",
    THIRD_DOZEN: "3rd 12",
    FIRST_18: "1 to 18",
    SECOND_18: "19 to 36",
    EVEN: "Even",
    ODD: "Odd",
    FIRST_COLUMN: "1st Column",
    SECOND_COLUMN: "2nd Column",
    THIRD_COLUMN: "3rd Column",
    RED: "Red",
    BLACK: "Black",
};
