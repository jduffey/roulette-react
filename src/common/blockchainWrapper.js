import { ethers } from "ethers";

const provider = new ethers.providers.JsonRpcProvider("http://localhost:8545");

// TODO: retrieve addresses from file generated by initialization script so they aren't hardcoded
const FIRST_PLAYER_ADDRESS = "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266";
const SECOND_PLAYER_ADDRESS = "0x70997970C51812dc3A010C7d01b50e0d17dc79C8";
const THIRD_PLAYER_ADDRESS = "0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC";
const HOUSE_ADDRESS = "0x90F79bf6EB2c4f870365E785982E1f101E93b906";
// Contract addresses are calculated from a hash of the deployer's address and nonce.
const TOKEN_CONTRACT_ADDRESS = "0x057ef64E23666F000b34aE31332854aCBd1c8544";
const ROULETTE_CONTRACT_ADDRESS = "0x261D8c5e9742e6f7f1076Fa1F560894524e19cad";

async function incrementTotalSpins() {
    const signer = provider.getSigner(HOUSE_ADDRESS);
    const incrementTotalSpinsAbi = new ethers.Contract(
        ROULETTE_CONTRACT_ADDRESS,
        ["function incrementTotalSpins()"],
        signer
    );
    const tx = await incrementTotalSpinsAbi.incrementTotalSpins();
    return tx;
}

async function getTotalSpins() {
    const signer = provider.getSigner(HOUSE_ADDRESS);
    const getTotalSpinsAbi = new ethers.Contract(
        ROULETTE_CONTRACT_ADDRESS,
        ["function getTotalSpins() public view returns (uint256)"],
        signer
    );
    const count = await getTotalSpinsAbi.getTotalSpins();
    return count;
}

async function getEthBalance(address) {
    const signer = provider.getSigner(address);
    const balance = await signer.getBalance();
    return ethers.utils.formatEther(balance);
}

// TODO leaving this for reference
// async function sendEther(provider, from, to, amount) {
//     const signer = provider.getSigner(from);
//     const tx = await signer.sendTransaction({
//         to: to,
//         value: ethers.utils.parseEther(amount)
//     });
//     return tx;
// }

async function getBlock() {
    const block = await provider.getBlock();
    return block;
}

async function getTokenBalance(address) {
    const signer = provider.getSigner(address);
    const token = new ethers.Contract(
        TOKEN_CONTRACT_ADDRESS,
        ["function balanceOf(address) view returns (uint)"],
        signer
    );
    const balance = await token.balanceOf(address);
    const formattedBalance = ethers.utils.formatEther(balance);
    return formattedBalance;
}

async function depositEthForTokens(from, amount) {
    const signer = provider.getSigner(from);
    const token = new ethers.Contract(
        TOKEN_CONTRACT_ADDRESS,
        ["function deposit() payable"],
        signer
    );
    const tx = await token.deposit({
        value: ethers.utils.parseEther(amount)
    });
    return tx;
}

async function redeemTokensForEth(from, amount) {
    const signer = provider.getSigner(from);
    const token = new ethers.Contract(
        TOKEN_CONTRACT_ADDRESS,
        ["function withdraw(uint)"],
        signer
    );
    const tx = await token.withdraw(ethers.utils.parseEther(amount));
    return tx;
}

async function transferFrom(from, to, amount) {
    const signer = provider.getSigner(from);
    const token = new ethers.Contract(
        TOKEN_CONTRACT_ADDRESS,
        ["function transferFrom(address, address, uint) returns (bool)"],
        signer
    );
    const tx = await token.transferFrom(from, to, ethers.utils.parseEther(amount));
    return tx;
}

async function incrementJackpotBalance(amount) {
    const signer = provider.getSigner(HOUSE_ADDRESS);
    const contract = new ethers.Contract(
        ROULETTE_CONTRACT_ADDRESS,
        ["function incrementJackpotBalance(uint256)"],
        signer
    );
    const tx = await contract.incrementJackpotBalance(ethers.utils.parseEther(amount));
    return tx;
}

async function getJackpotBalance() {
    const signer = provider.getSigner(HOUSE_ADDRESS);
    const contract = new ethers.Contract(
        ROULETTE_CONTRACT_ADDRESS,
        ["function getJackpotBalance() public view returns (uint256)"],
        signer
    );
    const balance = await contract.getJackpotBalance();
    const formattedBalance = ethers.utils.formatEther(balance);
    return formattedBalance;
}

let tokenSymbol;
(new ethers.Contract(
    TOKEN_CONTRACT_ADDRESS,
    ["function symbol() view returns (string)"],
    provider
)).symbol().then((symbol) => {
    tokenSymbol = symbol;
});

export {
    getEthBalance,
    getBlock,
    getTokenBalance,
    depositEthForTokens,
    redeemTokensForEth,
    transferFrom,
    incrementTotalSpins,
    getTotalSpins,
    incrementJackpotBalance,
    getJackpotBalance,
    FIRST_PLAYER_ADDRESS,
    SECOND_PLAYER_ADDRESS,
    THIRD_PLAYER_ADDRESS,
    HOUSE_ADDRESS,
    TOKEN_CONTRACT_ADDRESS,
    tokenSymbol,
};