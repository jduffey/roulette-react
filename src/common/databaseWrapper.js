const DATABASE_URL = new URL('/player', 'http://localhost:3001');

async function fetchPlayerBalance() {
    const res = await fetch(DATABASE_URL);
    if (!res.ok) {
        console.log("Not OK response from json-server: ", res);
        return `Error: ${res.status} ${res.statusText}`;
    }
    const json = await res.json();
    return json.balance;
}

async function updatePlayerBalance(balance) {
    const res = await fetch(DATABASE_URL, {
        method: 'PUT',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify({ balance }),
    });
    if (!res.ok) {
        console.log("Not OK response from json-server: ", res);
    }
}

export { fetchPlayerBalance, updatePlayerBalance };
