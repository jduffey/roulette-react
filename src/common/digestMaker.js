// https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest
export async function digestMessage(message) {
    const msgUint8 = new TextEncoder().encode(message);

    const hashBuffer = await crypto.subtle.digest('SHA-256', msgUint8);

    const hashArray = Array.from(new Uint8Array(hashBuffer));

    const hashHex = hashArray.map((b) => b.toString(16).padStart(2, '0')).join('');

    return hashHex;
}
