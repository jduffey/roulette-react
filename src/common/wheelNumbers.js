export const WHEEL_NUMBERS = {
    WN_0: "0",
    WN_00: "00",
    WN_1: "1",
    WN_2: "2",
    WN_3: "3",
    WN_4: "4",
    WN_5: "5",
    WN_6: "6",
    WN_7: "7",
    WN_8: "8",
    WN_9: "9",
    WN_10: "10",
    WN_11: "11",
    WN_12: "12",
    WN_13: "13",
    WN_14: "14",
    WN_15: "15",
    WN_16: "16",
    WN_17: "17",
    WN_18: "18",
    WN_19: "19",
    WN_20: "20",
    WN_21: "21",
    WN_22: "22",
    WN_23: "23",
    WN_24: "24",
    WN_25: "25",
    WN_26: "26",
    WN_27: "27",
    WN_28: "28",
    WN_29: "29",
    WN_30: "30",
    WN_31: "31",
    WN_32: "32",
    WN_33: "33",
    WN_34: "34",
    WN_35: "35",
    WN_36: "36",
}
