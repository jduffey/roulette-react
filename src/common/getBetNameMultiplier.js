import { BET_NAMES } from "./betNames";

export const getBetNameMultiplier = (betName) => {
    console.log("getBetNameMultiplier should be called once per bet placed", betName);
    switch (betName) {
        case BET_NAMES.STRAIGHT_UP_0:
        case BET_NAMES.STRAIGHT_UP_00:
        case BET_NAMES.STRAIGHT_UP_1:
        case BET_NAMES.STRAIGHT_UP_2:
        case BET_NAMES.STRAIGHT_UP_3:
        case BET_NAMES.STRAIGHT_UP_4:
        case BET_NAMES.STRAIGHT_UP_5:
        case BET_NAMES.STRAIGHT_UP_6:
        case BET_NAMES.STRAIGHT_UP_7:
        case BET_NAMES.STRAIGHT_UP_8:
        case BET_NAMES.STRAIGHT_UP_9:
        case BET_NAMES.STRAIGHT_UP_10:
        case BET_NAMES.STRAIGHT_UP_11:
        case BET_NAMES.STRAIGHT_UP_12:
        case BET_NAMES.STRAIGHT_UP_13:
        case BET_NAMES.STRAIGHT_UP_14:
        case BET_NAMES.STRAIGHT_UP_15:
        case BET_NAMES.STRAIGHT_UP_16:
        case BET_NAMES.STRAIGHT_UP_17:
        case BET_NAMES.STRAIGHT_UP_18:
        case BET_NAMES.STRAIGHT_UP_19:
        case BET_NAMES.STRAIGHT_UP_20:
        case BET_NAMES.STRAIGHT_UP_21:
        case BET_NAMES.STRAIGHT_UP_22:
        case BET_NAMES.STRAIGHT_UP_23:
        case BET_NAMES.STRAIGHT_UP_24:
        case BET_NAMES.STRAIGHT_UP_25:
        case BET_NAMES.STRAIGHT_UP_26:
        case BET_NAMES.STRAIGHT_UP_27:
        case BET_NAMES.STRAIGHT_UP_28:
        case BET_NAMES.STRAIGHT_UP_29:
        case BET_NAMES.STRAIGHT_UP_30:
        case BET_NAMES.STRAIGHT_UP_31:
        case BET_NAMES.STRAIGHT_UP_32:
        case BET_NAMES.STRAIGHT_UP_33:
        case BET_NAMES.STRAIGHT_UP_34:
        case BET_NAMES.STRAIGHT_UP_35:
        case BET_NAMES.STRAIGHT_UP_36:
            return 35;

        case BET_NAMES.FIRST_DOZEN:
        case BET_NAMES.SECOND_DOZEN:
        case BET_NAMES.THIRD_DOZEN:
        case BET_NAMES.FIRST_COLUMN:
        case BET_NAMES.SECOND_COLUMN:
        case BET_NAMES.THIRD_COLUMN:
            return 2;

        case BET_NAMES.FIRST_18:
        case BET_NAMES.SECOND_18:
        case BET_NAMES.EVEN:
        case BET_NAMES.ODD:
        case BET_NAMES.RED:
        case BET_NAMES.BLACK:
            return 1;

        default:
            throw new Error("Invalid bet name", betName);
    }
}
