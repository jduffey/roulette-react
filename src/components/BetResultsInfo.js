import { WINNING_CRITERIA } from "../common/winning-criteria";

const className = "bet-results-info";
export function BetResultsInfo(props) {
    const sumBetAmounts = Object.keys(props.bets).reduce((acc, betOption) => {
        const betAmountOnBet = props.bets[betOption];
        return acc + betAmountOnBet;
    }, 0);

    const sumWinnings = Object.keys(props.bets).reduce((acc, betOption) => {
        const winningsOnBet = calculateWinningsOnBet(props, betOption);
        return acc + winningsOnBet;
    }, 0);

    const sumBetsReturned = Object.keys(props.bets).reduce((acc, betOption) => {
        const betReturnedAmount = calculateBetReturnedAmount(props, betOption);
        return acc + betReturnedAmount;
    }, 0);

    const startingBalanceText = props.startingBalance ?
        "$ " + props.startingBalance.toString() :
        "";

    const netChangeInBalance = sumWinnings + sumBetsReturned - sumBetAmounts;
    const netChangeInBalanceText = props.startingBalance ?
        "$ " + netChangeInBalance.toString() :
        "";

    const finalBalance = props.startingBalance + netChangeInBalance;
    const finalBalanceText = props.startingBalance ?
        "$ " + finalBalance.toString() :
        "";

    return (
        <div className={className}>
            <div
                className="bet-info-table-title"
            >
                PREVIOUS ROUND RESULTS
            </div>
            <table className="bet-results-info-table">
                <tbody>
                    <tr style={{
                        textAlign: "center",
                    }}>
                        <th>Bet Name</th>
                        <th>Bet Amount</th>
                        <th>Winnings</th>
                        <th>Bet Returned</th>
                    </tr>
                    {Object.keys(props.bets).map((betOption) => {
                        const betAmountOnBet = props.bets[betOption];

                        let winningsOnBet = calculateWinningsOnBet(props, betOption);
                        const winningsOnBetText = winningsOnBet ? "$ " + winningsOnBet.toString() : "";

                        let betReturnedAmount = calculateBetReturnedAmount(props, betOption);
                        const betReturnedAmountText = betReturnedAmount ? "$ " + betReturnedAmount.toString() : "";

                        return (
                            <tr key={betOption}>
                                <td>{betOption}</td>
                                <td className="bet-results-info-table-bet-amount">{"$ " + betAmountOnBet.toString()}</td>
                                <td>{winningsOnBetText}</td>
                                <td>{betReturnedAmountText}</td>
                            </tr>
                        );
                    })}
                    <tr style={{ height: "10px" }}></tr>
                    <tr>
                        <td>TOTALS</td>
                        <td>{"$ " + sumBetAmounts}</td>
                        <td>{"$ " + sumWinnings}</td>
                        <td>{"$ " + sumBetsReturned}</td>
                    </tr>
                    <tr style={{ height: "10px" }}></tr>
                    <tr className="balance-change-value">
                        <td>
                            Starting
                        </td>
                        <td colSpan={3}>
                            {startingBalanceText}
                        </td>
                    </tr>
                    <tr className="balance-change-value">
                        <td>
                            Net
                        </td>
                        <td colSpan={3}>
                            {netChangeInBalanceText}
                        </td>
                    </tr>
                    <tr className="balance-change-value">
                        <td>
                            Final
                        </td>
                        <td colSpan={3}>
                            {finalBalanceText}
                        </td>
                    </tr>
                </tbody>
            </table>
        </div>
    );
}

function calculateWinningsOnBet(props, betOption) {
    const multipliers = {
        "1": 35,
        "2": 35,
        "3": 35,
        "4": 35,
        "5": 35,
        "6": 35,
        "7": 35,
        "8": 35,
        "9": 35,
        "10": 35,
        "11": 35,
        "12": 35,
        "13": 35,
        "14": 35,
        "15": 35,
        "16": 35,
        "17": 35,
        "18": 35,
        "19": 35,
        "20": 35,
        "21": 35,
        "22": 35,
        "23": 35,
        "24": 35,
        "25": 35,
        "26": 35,
        "27": 35,
        "28": 35,
        "29": 35,
        "30": 35,
        "31": 35,
        "32": 35,
        "33": 35,
        "34": 35,
        "35": 35,
        "36": 35,
        "0": 35,
        "00": 35,
        "1st 12": 2,
        "2nd 12": 2,
        "3rd 12": 2,
        "Top Row": 2,
        "Middle Row": 2,
        "Bottom Row": 2,
        "1 to 18": 1,
        "19 to 36": 1,
        "Even": 1,
        "Odd": 1,
        "Red": 1,
        "Black": 1,
    };

    const betAmount = props.bets[betOption];

    return WINNING_CRITERIA[props.winningWheelNumber].includes(betOption) ?
        betAmount * multipliers[betOption] :
        0;
}

function calculateBetReturnedAmount(props, betOption) {
    return WINNING_CRITERIA[props.winningWheelNumber].includes(betOption) ?
        props.bets[betOption] :
        0;
}
