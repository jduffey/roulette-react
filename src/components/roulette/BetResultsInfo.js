import { getWheelNumberColor } from "../../common/getWheelNumberColor";

const CLASS_NAME = "BetResultsInfo-component";
export function BetResultsInfo(props) {
    const previousRoundHasBetResults = props.previousRoundResults !== null;

    let startingBalanceText = "-";
    let finalBalanceText = "-";
    let netChangeInBalanceText = "-";
    let totalBetAmountText = "-";
    let totalWinningsText = "-";
    let totalBetsReturnedText = "-";
    if (previousRoundHasBetResults) {
        startingBalanceText = `$ ${props.previousRoundResults.startingBalance}`;
        finalBalanceText = `$ ${props.previousRoundResults.finalBalance}`;
        netChangeInBalanceText = `$ ${props.previousRoundResults.finalBalance - props.previousRoundResults.startingBalance}`;

        const sumBetAmounts = Object.values(props.previousRoundResults.resultsOfBets).reduce((acc, bet) => acc + bet.betAmount, 0);
        const sumWinnings = Object.values(props.previousRoundResults.resultsOfBets).reduce((acc, bet) => acc + bet.winningsOnBet, 0);
        const sumBetsReturned = Object.values(props.previousRoundResults.resultsOfBets).reduce((acc, bet) => acc + bet.betReturned, 0);

        totalBetAmountText = `$ ${sumBetAmounts}`;
        totalWinningsText = `$ ${sumWinnings}`;
        totalBetsReturnedText = `$ ${sumBetsReturned}`;
    }

    const previousWheelNumberDiv = () => {
        const backgroundColor = previousRoundHasBetResults ?
            getWheelNumberColor(props.previousRoundResults.winningWheelNumber)
            : "#dfdfdf";

        const wheelNumberText = previousRoundHasBetResults ?
            props.previousRoundResults.winningWheelNumber :
            "??";

        return (
            <div
                className="bet-info-table-title-spin-result"
                style={{ backgroundColor }}
            >
                {wheelNumberText}
            </div>
        )
    }

    const betResultsRows = () => Object.keys(props.previousRoundResults.resultsOfBets).map((betName) => {
        const betAmountOnBet = props.previousRoundResults.resultsOfBets[betName].betAmount;
        const winningsOnBet = props.previousRoundResults.resultsOfBets[betName].winningsOnBet;
        const betReturnedAmount = props.previousRoundResults.resultsOfBets[betName].betReturned;
        return (
            <tr key={betName}>
                <td>{betName}</td>
                <td className="bet-results-info-table-bet-amount">{`$ ${betAmountOnBet}`}</td>
                <td>{`$ ${winningsOnBet}`}</td>
                <td>{`$ ${betReturnedAmount}`}</td>
            </tr>
        );
    });

    return (
        <div
            className={CLASS_NAME}
        >
            <div
                className="bet-info-table-title-container"
            >
                <div
                    className="bet-info-table-title-text"
                >
                    PREVIOUS ROUND RESULTS
                </div>

                {previousWheelNumberDiv()}

            </div>
            <table className="bet-results-info-table">
                <tbody>
                    <tr style={{
                        textAlign: "center",
                    }}>
                        <th>Bet Name</th>
                        <th>Bet Amount</th>
                        <th>Winnings</th>
                        <th>Bet Returned</th>
                    </tr>

                    {previousRoundHasBetResults ? betResultsRows() : null}

                    <tr style={{ height: "10px" }} />
                    <tr>
                        <td>TOTALS</td>
                        <td>{totalBetAmountText}</td>
                        <td>{totalWinningsText}</td>
                        <td>{totalBetsReturnedText}</td>
                    </tr>
                    <tr style={{ height: "10px" }} />
                    <tr className="balance-change-value">
                        <td>
                            Starting
                        </td>
                        <td colSpan={3}>
                            {startingBalanceText}
                        </td>
                    </tr>
                    <tr className="balance-change-value">
                        <td>
                            Net
                        </td>
                        <td colSpan={3}>
                            {netChangeInBalanceText}
                        </td>
                    </tr>
                    <tr className="balance-change-value">
                        <td>
                            Final
                        </td>
                        <td colSpan={3}>
                            {finalBalanceText}
                        </td>
                    </tr>
                </tbody>
            </table>
        </div >
    );
}
