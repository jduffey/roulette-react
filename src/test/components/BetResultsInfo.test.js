import renderer from 'react-test-renderer';

import { BetResultsInfo } from '../../components/BetResultsInfo';

import { BET_NAMES } from '../../common/betNames';
import { WHEEL_NUMBERS } from '../../common/wheelNumbers';

describe('BetResultsInfo', () => {
    it('renders with no bets', () => {
        const sut =
            BetResultsInfo({
                bets: {},
            });

        const actual = renderer.create(sut);

        expect(actual).toMatchSnapshot();
    });

    it('renders with bets', () => {
        const sut =
            BetResultsInfo({
                bets: {
                    [BET_NAMES.EVEN]: 10,
                    [BET_NAMES.BLACK]: 20,
                    [BET_NAMES.FIRST_DOZEN]: 30,
                },
                winningWheelNumber: WHEEL_NUMBERS.WN_7,
                startingBalance: 1000,
            });

        const actual = renderer.create(sut);

        expect(actual).toMatchSnapshot();
    });

    it.each([
        [WHEEL_NUMBERS.WN_0, 'green'],
        [WHEEL_NUMBERS.WN_1, 'red'],
        [WHEEL_NUMBERS.WN_2, 'black'],
        [undefined, 'generic grey'],
    ])('renders winning wheel numbers with different color backgrounds (%s -> %s)', (winningWheelNumber, _bgColor) => {
        const sut =
            BetResultsInfo({
                bets: {},
                winningWheelNumber,
                startingBalance: 1000,
            });

        const actual = renderer.create(sut);

        expect(actual).toMatchSnapshot();
    });
});
