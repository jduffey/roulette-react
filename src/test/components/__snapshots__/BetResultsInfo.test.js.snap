// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetResultsInfo renders winning wheel numbers with different color backgrounds (0 -> green) 1`] = `
<div
  className="BetResultsInfo-component"
>
  <div
    className="bet-info-table-title-container"
  >
    <div
      className="bet-info-table-title-text"
    >
      PREVIOUS ROUND RESULTS
    </div>
    <div
      className="bet-info-table-title-spin-result"
      style={
        Object {
          "backgroundColor": "#016D29",
        }
      }
    >
      0
    </div>
  </div>
  <table
    className="bet-results-info-table"
  >
    <tbody>
      <tr
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <th>
          Bet Name
        </th>
        <th>
          Bet Amount
        </th>
        <th>
          Winnings
        </th>
        <th>
          Bet Returned
        </th>
      </tr>
      <tr
        style={
          Object {
            "height": "10px",
          }
        }
      />
      <tr>
        <td>
          TOTALS
        </td>
        <td>
          $ 0
        </td>
        <td>
          $ 0
        </td>
        <td>
          $ 0
        </td>
      </tr>
      <tr
        style={
          Object {
            "height": "10px",
          }
        }
      />
      <tr
        className="balance-change-value"
      >
        <td>
          Starting
        </td>
        <td
          colSpan={3}
        >
          $ 1000
        </td>
      </tr>
      <tr
        className="balance-change-value"
      >
        <td>
          Net
        </td>
        <td
          colSpan={3}
        >
          $ 0
        </td>
      </tr>
      <tr
        className="balance-change-value"
      >
        <td>
          Final
        </td>
        <td
          colSpan={3}
        >
          $ 1000
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`BetResultsInfo renders winning wheel numbers with different color backgrounds (1 -> red) 1`] = `
<div
  className="BetResultsInfo-component"
>
  <div
    className="bet-info-table-title-container"
  >
    <div
      className="bet-info-table-title-text"
    >
      PREVIOUS ROUND RESULTS
    </div>
    <div
      className="bet-info-table-title-spin-result"
      style={
        Object {
          "backgroundColor": "#d94848",
        }
      }
    >
      1
    </div>
  </div>
  <table
    className="bet-results-info-table"
  >
    <tbody>
      <tr
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <th>
          Bet Name
        </th>
        <th>
          Bet Amount
        </th>
        <th>
          Winnings
        </th>
        <th>
          Bet Returned
        </th>
      </tr>
      <tr
        style={
          Object {
            "height": "10px",
          }
        }
      />
      <tr>
        <td>
          TOTALS
        </td>
        <td>
          $ 0
        </td>
        <td>
          $ 0
        </td>
        <td>
          $ 0
        </td>
      </tr>
      <tr
        style={
          Object {
            "height": "10px",
          }
        }
      />
      <tr
        className="balance-change-value"
      >
        <td>
          Starting
        </td>
        <td
          colSpan={3}
        >
          $ 1000
        </td>
      </tr>
      <tr
        className="balance-change-value"
      >
        <td>
          Net
        </td>
        <td
          colSpan={3}
        >
          $ 0
        </td>
      </tr>
      <tr
        className="balance-change-value"
      >
        <td>
          Final
        </td>
        <td
          colSpan={3}
        >
          $ 1000
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`BetResultsInfo renders winning wheel numbers with different color backgrounds (2 -> black) 1`] = `
<div
  className="BetResultsInfo-component"
>
  <div
    className="bet-info-table-title-container"
  >
    <div
      className="bet-info-table-title-text"
    >
      PREVIOUS ROUND RESULTS
    </div>
    <div
      className="bet-info-table-title-spin-result"
      style={
        Object {
          "backgroundColor": "#222222",
        }
      }
    >
      2
    </div>
  </div>
  <table
    className="bet-results-info-table"
  >
    <tbody>
      <tr
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <th>
          Bet Name
        </th>
        <th>
          Bet Amount
        </th>
        <th>
          Winnings
        </th>
        <th>
          Bet Returned
        </th>
      </tr>
      <tr
        style={
          Object {
            "height": "10px",
          }
        }
      />
      <tr>
        <td>
          TOTALS
        </td>
        <td>
          $ 0
        </td>
        <td>
          $ 0
        </td>
        <td>
          $ 0
        </td>
      </tr>
      <tr
        style={
          Object {
            "height": "10px",
          }
        }
      />
      <tr
        className="balance-change-value"
      >
        <td>
          Starting
        </td>
        <td
          colSpan={3}
        >
          $ 1000
        </td>
      </tr>
      <tr
        className="balance-change-value"
      >
        <td>
          Net
        </td>
        <td
          colSpan={3}
        >
          $ 0
        </td>
      </tr>
      <tr
        className="balance-change-value"
      >
        <td>
          Final
        </td>
        <td
          colSpan={3}
        >
          $ 1000
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`BetResultsInfo renders winning wheel numbers with different color backgrounds (undefined -> generic grey) 1`] = `
<div
  className="BetResultsInfo-component"
>
  <div
    className="bet-info-table-title-container"
  >
    <div
      className="bet-info-table-title-text"
    >
      PREVIOUS ROUND RESULTS
    </div>
    <div
      className="bet-info-table-title-spin-result"
      style={
        Object {
          "backgroundColor": "#dfdfdf",
        }
      }
    >
      ?
    </div>
  </div>
  <table
    className="bet-results-info-table"
  >
    <tbody>
      <tr
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <th>
          Bet Name
        </th>
        <th>
          Bet Amount
        </th>
        <th>
          Winnings
        </th>
        <th>
          Bet Returned
        </th>
      </tr>
      <tr
        style={
          Object {
            "height": "10px",
          }
        }
      />
      <tr>
        <td>
          TOTALS
        </td>
        <td>
          $ 0
        </td>
        <td>
          $ 0
        </td>
        <td>
          $ 0
        </td>
      </tr>
      <tr
        style={
          Object {
            "height": "10px",
          }
        }
      />
      <tr
        className="balance-change-value"
      >
        <td>
          Starting
        </td>
        <td
          colSpan={3}
        >
          $ 1000
        </td>
      </tr>
      <tr
        className="balance-change-value"
      >
        <td>
          Net
        </td>
        <td
          colSpan={3}
        >
          $ 0
        </td>
      </tr>
      <tr
        className="balance-change-value"
      >
        <td>
          Final
        </td>
        <td
          colSpan={3}
        >
          $ 1000
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`BetResultsInfo renders with bets 1`] = `
<div
  className="BetResultsInfo-component"
>
  <div
    className="bet-info-table-title-container"
  >
    <div
      className="bet-info-table-title-text"
    >
      PREVIOUS ROUND RESULTS
    </div>
    <div
      className="bet-info-table-title-spin-result"
      style={
        Object {
          "backgroundColor": "#d94848",
        }
      }
    >
      7
    </div>
  </div>
  <table
    className="bet-results-info-table"
  >
    <tbody>
      <tr
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <th>
          Bet Name
        </th>
        <th>
          Bet Amount
        </th>
        <th>
          Winnings
        </th>
        <th>
          Bet Returned
        </th>
      </tr>
      <tr>
        <td>
          Even
        </td>
        <td
          className="bet-results-info-table-bet-amount"
        >
          $ 10
        </td>
        <td />
        <td />
      </tr>
      <tr>
        <td>
          Black
        </td>
        <td
          className="bet-results-info-table-bet-amount"
        >
          $ 20
        </td>
        <td />
        <td />
      </tr>
      <tr>
        <td>
          1st 12
        </td>
        <td
          className="bet-results-info-table-bet-amount"
        >
          $ 30
        </td>
        <td>
          $ 60
        </td>
        <td>
          $ 30
        </td>
      </tr>
      <tr
        style={
          Object {
            "height": "10px",
          }
        }
      />
      <tr>
        <td>
          TOTALS
        </td>
        <td>
          $ 60
        </td>
        <td>
          $ 60
        </td>
        <td>
          $ 30
        </td>
      </tr>
      <tr
        style={
          Object {
            "height": "10px",
          }
        }
      />
      <tr
        className="balance-change-value"
      >
        <td>
          Starting
        </td>
        <td
          colSpan={3}
        >
          $ 1000
        </td>
      </tr>
      <tr
        className="balance-change-value"
      >
        <td>
          Net
        </td>
        <td
          colSpan={3}
        >
          $ 30
        </td>
      </tr>
      <tr
        className="balance-change-value"
      >
        <td>
          Final
        </td>
        <td
          colSpan={3}
        >
          $ 1030
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`BetResultsInfo renders with no bets 1`] = `
<div
  className="BetResultsInfo-component"
>
  <div
    className="bet-info-table-title-container"
  >
    <div
      className="bet-info-table-title-text"
    >
      PREVIOUS ROUND RESULTS
    </div>
    <div
      className="bet-info-table-title-spin-result"
      style={
        Object {
          "backgroundColor": "#dfdfdf",
        }
      }
    >
      ?
    </div>
  </div>
  <table
    className="bet-results-info-table"
  >
    <tbody>
      <tr
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <th>
          Bet Name
        </th>
        <th>
          Bet Amount
        </th>
        <th>
          Winnings
        </th>
        <th>
          Bet Returned
        </th>
      </tr>
      <tr
        style={
          Object {
            "height": "10px",
          }
        }
      />
      <tr>
        <td>
          TOTALS
        </td>
        <td>
          $ 0
        </td>
        <td>
          $ 0
        </td>
        <td>
          $ 0
        </td>
      </tr>
      <tr
        style={
          Object {
            "height": "10px",
          }
        }
      />
      <tr
        className="balance-change-value"
      >
        <td>
          Starting
        </td>
        <td
          colSpan={3}
        />
      </tr>
      <tr
        className="balance-change-value"
      >
        <td>
          Net
        </td>
        <td
          colSpan={3}
        />
      </tr>
      <tr
        className="balance-change-value"
      >
        <td>
          Final
        </td>
        <td
          colSpan={3}
        />
      </tr>
    </tbody>
  </table>
</div>
`;
